pygeoapi/provider/filesystem.py

```diff
@@ -219,76 +224,89 @@ def _describe_file(filepath):
         'properties': {}
     }
 
+    pth = os.path.splitext(filepath)[0]
     try:
-        import rasterio
-        from rasterio.crs import CRS
-        from rasterio.warp import transform_bounds
-        LOGGER.warning('rasterio not found. Cannot derive geospatial properties')  # noqa
-    except ImportError as err:
-        LOGGER.warning(err)
-        return content
-
-    try:
-        import fiona
-    except ImportError as err:
-        LOGGER.warning('fiona not found. Cannot derive geospatial properties')
-        LOGGER.warning(err)
-        return content
-
-    try:  # raster
-        LOGGER.debug('Testing raster data detection')
-        d = rasterio.open(filepath)
-        content['bbox'] = [
-            d.bounds.left,
-            d.bounds.bottom,
-            d.bounds.right,
-            d.bounds.top
-        ]
-        content['geometry'] = {
-            'type': 'Polygon',
-            'coordinates': [[
-                [d.bounds.left, d.bounds.bottom],
-                [d.bounds.left, d.bounds.top],
-                [d.bounds.right, d.bounds.top],
-                [d.bounds.right, d.bounds.bottom],
-                [d.bounds.left, d.bounds.bottom]
-            ]]
-        }
-        for k, v in d.tags(1).items():
-            content['properties'][k] = v
-    except rasterio.errors.RasterioIOError:
-        LOGGER.debug('Testing vector data detection')
-        d = fiona.open(filepath)
-        scrs = CRS(d.crs)
-        if scrs.to_epsg() is not None and scrs.to_epsg() != 4326:
-            tcrs = CRS.from_epsg(4326)
-            bnds = transform_bounds(scrs, tcrs,
-                                    d.bounds[0], d.bounds[1],
-                                    d.bounds[2], d.bounds[3])
-            content['properties']['projection'] = scrs.to_epsg()
-        else:
-            bnds = d.bounds
-
-        if d.schema['geometry'] not in [None, 'None']:
+        md = read_mcf('./{}.yml'.format(pth))
```

Comment:
Instead of try/except, can we test if the file is there? Like:

```python
mcf_file = './{}.yml'.format(pth)

if os.path.isfile(mcf_file):
    LOGGER.debug('Found possible MCF metadata file')
    try:
        LOGGER.debug('Parsing {}'.format(mcf_file))
        md = read_mcf(mcf_file)
        stacjson = STACItemOutputSchema.write(STACItemOutputSchema, md) 
        stacdata = loads(stacjson)
        for k, v in stacdata.items():
            content[k] = v
    except MCFReadError as err:
         LOGGER.warning('MCF error: ' + str(err))
```

In other words, we should try to detect if the given MCF file is on disk before we try to parse.  This will also negate the need for the `OSError` as a result.